pkg/k8sutils/secret.go

```diff
@@ -17,3 +20,26 @@ func SecretStringDataFromData(secret *v1.Secret) map[string]string {
 	}
 	return stringData
 }
+
+func IsSecretWatchedByApicast(client k8sclient.Client, secretName, secretNamespace string) bool {
+	secret := &v1.Secret{}
+	objKey := k8sclient.ObjectKey{
+		Name:      secretName,
+		Namespace: secretNamespace,
+	}
+
+	err := client.Get(context.TODO(), objKey, secret)
+	if err != nil {
+		return false
+	}
+
+	existingLabels := secret.Labels
+
+	if existingLabels != nil {
+		if _, ok := existingLabels["apicast.apps.3scale.net/watched-by"]; ok {
```

Comment:
can we have a apicast operator and 3scale operator on a single namespace where 3scale operator watches secret X and apicast X1, will watch-by trigger both operators then?